plugins/pvio/pvio_socket.c

```diff
@@ -786,7 +786,13 @@ my_bool pvio_socket_connect(MARIADB_PVIO *pvio, MA_PVIO_CINFO *cinfo)
     else
 #endif
     {
-      strcpy(UNIXaddr.sun_path, cinfo->unix_socket);
+      size_t sun_path_size = sizeof(UNIXaddr.sun_path);
+      strncpy(UNIXaddr.sun_path, cinfo->unix_socket, sun_path_size - 1);
+      if (sun_path_size == strlen(UNIXaddr.sun_path) + 1 && UNIXaddr.sun_path[sun_path_size - 1] != '\0')
+      {
+        // Making the string null-terminated
+        UNIXaddr.sun_path[sun_path_size] = '\0';
```

Comment:
shouldn't that be [sun_path_size - 1] ?

Reply:
Yep, I've already fixed it.

Just haven't time to update it

Reply:
Done